db/silkworm/db/access_layer.cpp

```diff
@@ -204,17 +203,11 @@ static std::optional<ByteView> historical_storage(lmdb::Transaction& txn, const
                                                   uint64_t incarnation, const evmc::bytes32& location,
                                                   uint64_t block_number) {
     auto history_table{txn.open(table::kStorageHistory)};
-    std::optional<Entry> entry{history_table->seek(storage_history_key(address, location, block_number))};
+    std::optional<Entry> entry{history_table->seek(account_history_key(address, block_number))};
```

Comment:
The changes to `historical_storage` don't look right. Storage history key is different from that of account history.